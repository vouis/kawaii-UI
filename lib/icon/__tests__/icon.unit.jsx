import Icon from '../index'
import renderer from 'react-test-renderer'
import * as React from "react";
import { mount } from 'enzyme';

describe('Icon', () => {
    it('input icon name', () => {
        const result = renderer.create(
            <Icon name='planet'/>
        ).toJSON()
        expect(result).toMatchSnapshot()
    })
    it('icon event', () => {
        const fn = jest.fn()
        const component = mount(<Icon name='planet' onClick={fn}/>)
        component.find('svg').simulate('click')
        expect(fn).toBeCalled()

    })
});