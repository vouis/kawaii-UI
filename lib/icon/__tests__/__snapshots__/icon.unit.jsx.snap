// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon input icon name 1`] = `
<div>
  <svg
    className="kawaii_icon"
  >
    <use
      xlinkHref="#planet"
    />
  </svg>
</div>
`;
